components/pkg-export-helm/src/maintainer.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2018 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::str::FromStr;
+use std::result;
+use clap::ArgMatches;
+use url::Url;
+
+use export_docker::Result;
+
+use error::Error;
+
+#[derive(Clone, Debug)]
+pub struct Maintainer {
+    name: String,
+    email: Option<String>,
+    url: Option<String>,
+}
+
+impl Maintainer {
+    pub fn from_args(matches: &ArgMatches) -> Result<Vec<Self>> {
+        let mut maintainers = Vec::new();
+
+        if let Some(args) = matches.values_of("MAINTAINER") {
+            for arg in args {
```

Comment:
Out of curiosity, did you try doing this with `map`/`collect`?

Reply:
Not sure but I can try..

Reply:
Actually same issue here. `arg.parse` can end up in error, it can't be execute as part of a map.